crates/ide/src/lib.rs

```diff
@@ -405,7 +405,30 @@ impl Analysis {
         self.with_db(|db| {
             symbol_index::world_symbols(db, query)
                 .into_iter() // xx: should we make this a par iter?
-                .filter_map(|s| s.def.try_to_nav(db))
+                .filter_map(|s| {
+                    let nav = s.def.try_to_nav(db);
+
+                    match nav {
+                        Some(mut inner) => {
+                            let canpath = s.def.canonical_path(db);
+                            // With the use of doc aliases, we should adopt `FileSymbol`s name.
+                            // Otherwise aliases would be ignored.
+                            if inner.name != s.name {
+                                inner.name = SmolStr::from(format!(
+                                    "{} ( see `{}` )",
+                                    s.name,
+                                    if canpath.is_some() {
+                                        canpath.unwrap()
+                                    } else {
+                                        inner.name.to_string()
+                                    }
+                                ));
+                            }
+                            Some(inner)
+                        }
+                        None => None,
+                    }
+                })
```

Comment:
This is the wrong place to do this I think, and not the way we should handle it. What I would propose is to add a field to `FileSymbol` to mark whether it is from an alias or not.

Then in https://github.com/rust-lang/rust-analyzer/blob/bb78059be4d090571bd70de57831f84eee5be678/crates/rust-analyzer/src/handlers/request.rs#L518-L534
We should fix up the `SymbolInformation::name` field if the alias flag is set. I'd also propose to change the text to `"{aliasname} (alias `{path}` )"`

Reply:
I find your approach of editing `FileSymbol` much better. However during the process of creating a `SymbolInformation` , a `FileSymbol` has to go through becoming a `NavigationTarget` first and then this is then turned into a `SymbolInformation`. `NavigationTarget` completely disregards the fact that its name may stem from an alias and modifying `NavigationTarget` is not an easy job. The other option is to make ( possibly expensive ) calls that would turn `FileId` of  `NavigationTarget` back to the `FileSymbol` we are looking for ( for which I have no idea how I can do it). My next commit will change the content of `NavigationTarget` as you wished but I think you won't be happy also that I had to change so much.  :)